### ISSUE TITLE
Quarry supplies faulty fakedigger

### ISSUE THREAD

**Issue by thomasrudin** [minetest-mods/technic#447]
Stacktrace:
```
minetest_1                 | 2018-09-26 18:34:10: ERROR[Main]: 	[C]: in function 'error'
minetest_1                 | 2018-09-26 18:34:10: ERROR[Main]: 	/usr/local/bin/../builtin/game/misc.lua:11: in function 'check_player_privs'
minetest_1                 | 2018-09-26 18:34:10: ERROR[Main]: 	...al/bin/../games/minetest_game/mods/default/functions.lua:547: in function 'can_interact_with_node'
minetest_1                 | 2018-09-26 18:34:10: ERROR[Main]: 	/data/world//worldmods/technic/technic_chests/register.lua:307: in function 'can_dig'
minetest_1                 | 2018-09-26 18:34:10: ERROR[Main]: 	.../world//worldmods/technic/technic/machines/HV/quarry.lua:158: in function 'technic_run'
minetest_1                 | 2018-09-26 18:34:10: ERROR[Main]: 	...worldmods/technic/technic/machines/switching_station.lua:294: in function 'run_nodes'
minetest_1                 | 2018-09-26 18:34:10: ERROR[Main]: 	...worldmods/technic/technic/machines/switching_station.lua:300: in function <...worldmods/technic/technic/machines/switching_station.lua:235>
m
```

The minetest engine does not like the fake digger it seems:
https://github.com/minetest-mods/technic/blob/master/technic/machines/HV/quarry.lua#L149

**Minetest engine**: 0.4.17.1
**Minetest game**: 0.4-backport

**Comment by 12Me21** [comment]
Looks like a problem with the can_interact_with_node function.
```Lua
function default.can_interact_with_node(player, pos)
	if player then
		if minetest.check_player_privs(player, "protection_bypass") then
			return true
		end
	else
		return false
	end

	local meta = minetest.get_meta(pos)
	local owner = meta:get_string("owner")

	if not owner or owner == "" or owner == player:get_player_name() then
		return true
	end
```
The function clearly expects a Player (not just the name) because it uses `player:get_player_name()`
and yet it passes `player` directly to `check_player_privs` which expects a *name*.
```Lua
function core.check_player_privs(name, ...)
	if core.is_player(name) then
		name = name:get_player_name()
	elseif type(name) ~= "string" then
		error("core.check_player_privs expects a player or playername as " ..
			"argument.", 2)
	end
```

**Comment by 12Me21** [comment]
Making fake digger's `is_player` function return `true` would fix this, but there must be a reason for not doing that in the first place...

**Comment by 12Me21** [comment]
Perhaps the quarry should use pipeworks' fake player system (since technic depends on pipeworks anyway)

**Comment by thomasrudin** [comment]
fixed by #453
